pkg/pillar/pubsub/pubsub.go

```diff
@@ -62,6 +64,30 @@ type PubSub struct {
 	updaterList *Updaters
 }
 
+// PublicationOptions defines all the possible options a new publication may have
+type PublicationOptions struct {
+	AgentName  string
+	AgentScope string
+	TopicType  interface{}
+	Persistent bool
+}
+
+// NewPublication creates a new publication with given options
+func NewPublication(options PublicationOptions) (Publication, error) {
+	defaultPubSub := New(&socketdriver.SocketDriver{})
+	if options.Persistent && options.AgentName != "" && options.TopicType != nil {
+		return defaultPubSub.PublishPersistent(options.AgentName, options.TopicType)
```

Comment:
Do we really need these three? Or can we simplify them as well:

1) defaultPubSub.PublishPersistent
2) defaultPubSub.PublishScope
3) defaultPubSub.Publish

Or does it make sense to retain the three?